packages/devtools_app/lib/src/profiler/cpu_profile_controller.dart

```diff
@@ -34,6 +36,20 @@ class CpuProfilerController with SearchControllerMixin<CpuStackFrame> {
       _selectedCpuStackFrameNotifier;
   final _selectedCpuStackFrameNotifier = ValueNotifier<CpuStackFrame>(null);
 
+  /// Notifies that a user tag filter is set on the CPU profile flame chart.
+  ValueListenable<String> get userTagFilter => _userTagFilter;
+  final _userTagFilter = ValueNotifier<String>(userTagNone);
+
+  final _dataByTag = <String, CpuProfileData>{};
+
+  Iterable<String> get userTags => _dataByTag[userTagNone]?.userTags ?? [];
+
+  int selectedProfilerTabIndex = 0;
```

Comment:
why isn't this a value listenable?

Reply:
This isn't a value listenable because nothing needs to be rebuild when this changes. We are already rebuilding when the _tabController has changes. `selectedProfilerTabIndex` is only accessed by _CpuProfilerState one time in initState, so we don't need to listen to changes to it.